lib/schema/json_shemata.ex

```diff
@@ -7,62 +7,63 @@ defmodule ApplicationRunner.JsonSchemata do
 
   # Client (api)
   @component_api_directory "priv/components-api/api"
+  @component_api_root_file "component.schema.json"
 
   def get_schema_map(path) do
     GenServer.call(__MODULE__, {:get_schema_map, path})
   end
 
-  def load_raw_schema(schema, component_name) do
-    GenServer.call(__MODULE__, {:load_raw_schema, schema, component_name})
-  end
-
   def start_link(_) do
     GenServer.start_link(__MODULE__, %{}, name: __MODULE__)
   end
 
   # Server (callbacks)
   @impl true
   def init(_) do
-    root_json_directory = Application.app_dir(:application_runner, @component_api_directory)
-
-    relative_shemata_path =
-      Path.join(root_json_directory, "/**/*.schema.json")
-      |> Path.wildcard()
-      |> Enum.map(&Path.relative_to(&1, root_json_directory))
+    root_json_directory =
+      Application.app_dir(:application_runner, @component_api_directory) <>
+        "/" <> @component_api_root_file
 
-    schemata = Enum.map(relative_shemata_path, &load_schema/1)
-    schemata_map = Enum.zip(relative_shemata_path, schemata) |> Enum.into(%{})
+    {:ok, file_content} = File.read(root_json_directory)
 
-    {:ok, schemata_map}
-  end
-
-  def load_schema(path) do
-    schema =
-      read_schema(path)
+    schemata_map =
+      file_content
+      |> Jason.decode!()
       |> ExComponentSchema.Schema.resolve()
 
-    schema_properties = ApplicationRunner.SchemaParser.parse(schema)
+    schema = load_schema(schemata_map)
 
-    Map.merge(%{schema: schema}, schema_properties)
-  rescue
-    e in ExComponentSchema.Schema.InvalidSchemaError ->
-      reraise ExComponentSchema.Schema.InvalidSchemaError,
-              [message: "#{path} #{e.message}"],
-              __STACKTRACE__
+    {:ok, %{schemata_map: schemata_map, schema: schema}}
   end
 
-  defp load_raw_schema(schema, schemata_map, component_name) do
-    resolved_schema = ExComponentSchema.Schema.resolve(schema)
-
-    properties = ApplicationRunner.SchemaParser.parse(resolved_schema)
-
-    [{get_component_path(component_name), Map.merge(%{schema: resolved_schema}, properties)}]
-    |> Enum.into(schemata_map)
+  def load_schema(root_schema) do
+    Map.replace(
+      root_schema,
+      :refs,
+      Map.map(root_schema.refs, fn {id, ref} ->
```

Comment:
`Map.map` is deprecated use this instead : 

```suggestion
      Map.new(root_schema.refs, fn {id, ref} ->
```